### ISSUE TITLE
Problem with custom rule set provider

### ISSUE THREAD

**Issue by SumeraMartin** [JLLeitschuh/ktlint-gradle#697]
Hi, I updated ktlint o version 0.50. and ktilint-gradle to 11.5.0

I the past I was using custom rule set which worked fine but then it suddenly stopped to work (not sure which version). I updated API to new ktlint API which is using RuleSetProviderV3 for example. When I run tests in this module then it works fine. But these rules are ignored when I try to run ktlintCheck task.

Not sure if it is related but I found that file KtLintInvocation50 is using only one specific ruleset but in previous versions of this file were custom providers loaded. Is it intentionally due some limitations or it is just bug? Do you think that it could be fixed somehow? Is there a way of how can I help with it? Thanks

**Comment by wakingrufus** [comment]
Yeah i think you are right. I need to wire up the ruleProviders to load using ServiceProvider like we used to do.
Ill also add a test to make sure that this feature is exercised for all ktlintVersions to ensure we dont miss this again.

**Comment by wakingrufus** [comment]
I have a potential fix, but I am still working out how to test this properly

**Comment by wakingrufus** [comment]
This should be fixed in 11.5.1. Please try again

**Comment by SumeraMartin** [comment]
It works, thank you very much ❤️
